utils/codepointtrie/src/codepointtrie.rs

```diff
@@ -797,6 +775,59 @@ impl<'trie, T: TrieValue + Into<u32>> CodePointTrie<'trie, T> {
             cpm_range: init_range,
         }
     }
+
+    /// Returns a [`UnicodeSet`] for the code points that have the given
+    /// [`TrieValue`] in the trie.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use icu_codepointtrie::planes;
+    ///
+    /// let trie = planes::get_planes_trie();
+    ///
+    /// let plane_val = 2;
+    /// let sip = trie.get_set_for_value(plane_val as u8);
+    ///
+    /// let start = plane_val * 0x1_0000;
+    /// let end = start + 0xffff;
+    ///
+    /// assert!(!sip.contains_u32(start - 1));
+    /// assert!(sip.contains_u32(start));
+    /// assert!(sip.contains_u32(end));
+    /// assert!(!sip.contains_u32(end + 1));
+    /// ```
+    pub fn get_set_for_value(&self, value: T) -> UnicodeSet<'static> {
```

Comment:
As with Script_Extensions, please have two functions: one that returns a range iterator, and one that returns the set.  You have `.iter_ranges()` which returns range-value pairs; basically this function should just convert that iterator to one that has ranges without values.